assets/pages/oneoff-contributions/helpers/ajax.js

```diff
@@ -92,18 +92,18 @@ function postToEndpoint(request: Object, dispatch: Function, abParticipations: P
       dispatch(checkoutSuccess());
     }
     return response.json();
-  }).then((responseJson) => {
-    if (responseJson.error.exceptionType === 'CardException') {
-      dispatch(checkoutError('Your card has been declined.'));
-    } else {
-      const errorHttpCode = responseJson.error.errorCode || 'unknown';
-      const exceptionType = responseJson.error.exceptionType || 'unknown';
-      const errorName = responseJson.error.errorName || 'unknown';
-      logException(`Stripe payment attempt failed with following error: code: ${errorHttpCode} type: ${exceptionType} error-name: ${errorName}.`);
-      dispatch(checkoutError('There was an error processing your payment. Please\u00a0try\u00a0again\u00a0later.'));
+  }).then((data) => {
+    if (data.type === 'error') {
```

Comment:
If this condition evaluates to false, then we won't dispatch a checkout error; and if we haven't previously dispatched a checkout success, then this will result in no events being dispatched (which isn't good). With the current error model, this case should never occur, but I think it would be good to guard against it anyhow.

Reply:
Agreed!